modules/storages/app/services/projects/copy/file_links_dependent_service.rb

```diff
@@ -118,12 +118,26 @@ def files_info_query(storage_requests:, file_ids:)
         .result
     end
 
+    def remove_trailing_slashes_from_hash_keys(original_hash)
+      # Remove trailing slashes of directory paths.
+      # Nextcloud WebDAV PROPFIND requests for return for forlders location paths with trailing slashes.
+      # The `filesinfo` endpoint of the `OpenProject integration` Nextcloud app does not have those
+      # trailing slashes for folder paths. For matching both types of paths, we need to remove the trailing slashes.
+      clean_hash = {}
+      original_hash.each do |key, value|
+        clean_hash[key.chomp('/')] = value
+      end
+
+      clean_hash
+    end
+
     def folder_files_file_ids_deep_query(storage_requests:, path:)
-      storage_requests
-        .folder_files_file_ids_deep_query
-        .call(path:)
-        .on_failure { |r| add_error!("folder_files_file_ids_deep_query", r.to_active_model_errors) }
-        .result
+      result = storage_requests
+                 .folder_files_file_ids_deep_query
+                 .call(path:)
+                 .on_failure { |r| add_error!("folder_files_file_ids_deep_query", r.to_active_model_errors) }
+                 .result
+      remove_trailing_slashes_from_hash_keys(result)
```

Comment:
🔴 This is not the right location to fix it. One hint is, that you have added a comment describing nextcloud behaviour.

The sanitation of response data to fit the expected data must be done by the query. And we should add a unit test to the query to check, that the data is in the expected format.

Reply:
I get what you say and I agree. Two thoughts:
- Having the fix here solves a problem that we imminently encounter only here: It fixes the copy of the projects. It is a working patch for 13.0. We could release it and help our users and customers and support team.
- If we move it further upwards to the query, I do not know the side effects. The QA gets complex. Released as a patch it might bring new bugs (as other places might rely on the trailing slashes?...for me that is hard to judge).

So we could leave the code where it is, create a CODE MAINTENANCE ticket for 13.1 and have it go through a propper QA. Patch level releases should not bring bugs. Mayor and Minor releases always bring some bugs (however hard we try).

Different topic:
The code in this file has a very long method `copy_dependency` that does a lot of different stuff, one after the other. It was hard to understand what every section is doing. Having separate methods for copying automatically managed folders and manually managed folders would be start. Some method names are not really helpful (What is `folder_files_file_ids_deep_query`? Why is it named like that? If we have such unclear names then at least I would need a comment.)

Reply:
I moved this logic to `files_info_query`. So, the logic is the following: we save the slash last if it is directory. Doing so will be in sync with the behavior of `propfind_query` where we are not able to distinguish directories easily if we remove all last slashes.
As well I modified  `Storages::Peripherals::StorageInteraction::Nextcloud::Util.escape_path` to not eat the last slash if `path` has it at the end. So, it has more clear behavior now.